tldw_Server_API/app/core/Slides/visual_style_generation.py

```
@@ -5,68 +5,129 @@
+def _format_fallback_policy(fallback_policy: dict[str, Any]) -> str:
+    if not fallback_policy:
+        return "Keep slides readable in plain markdown if a structured block does not fit."
+
+    parts: list[str] = []
+    if "mode" in fallback_policy and str(fallback_policy["mode"]).strip():
+        parts.append(f"mode={fallback_policy['mode']}")
+    if "preserve_key_stats" in fallback_policy:
+        parts.append(f"preserve_key_stats={fallback_policy['preserve_key_stats']}")
+
+    extra_keys = sorted(
+        key for key in fallback_policy.keys() if key not in {"mode", "preserve_key_stats"}
```

Ruff (0.15.7)

[warning] Use `key in dict` instead of `key in dict.keys()`

Remove `.keys()`

(SIM118)